### ISSUE TITLE
Strange behaviour: Closing subscription of a Subject when call .subscribe() of a ObservablePromise

### ISSUE THREAD

**Issue by greenarr0w** [ReactiveX/rxjs#2507]
I have found a strange behaviour: 

**RxJS version:**
5.2.0 (on node v6.3.1)

**Code to reproduce:**
```javascript
let rxjs = require("rxjs");

let subject = new rxjs.Subject();
let subscriptionOfSubject = subject.subscribe();
subscriptionOfSubject.add(() => {
    console.log("unsubscribing");
});

let deferedPromise1 = Promise.defer();

rxjs.Observable.fromPromise(deferedPromise1.promise).subscribe(); // Does this cause my problem?

deferedPromise1.resolve();
console.log("is subscriptionOfSubject closed? " + subscriptionOfSubject.closed);

setTimeout(() => {
    console.log("is subscriptionOfSubject closed? " + subscriptionOfSubject.closed)
}, 5000);
```
The output of this is: 
is subscriptionOfSubject closed? false
unsubscribing
is subscriptionOfSubject closed? **true**

Why is the Subject unsubscribing, when i'm resolving a promise? They have nothing to do together.
The strange thing is, when I add a callback to the subscribe function of the PromiseObservable, then it's behaving correctly:

```javascript
let rxjs = require("rxjs");

let subject = new rxjs.Subject();
let subscriptionOfSubject = subject.subscribe();
subscriptionOfSubject.add(() => {
    console.log("unsubscribing");
});

let deferedPromise2 = Promise.defer();

// If I do this it works
rxjs.Observable.fromPromise(deferedPromise2.promise).subscribe(() => {
    console.log("on resolve");
});

deferedPromise2.resolve();
console.log("is subscriptionOfSubject closed? " + subscriptionOfSubject.closed);

setTimeout(() => {
    console.log("is subscriptionOfSubject closed? " + subscriptionOfSubject.closed)
}, 5000);
```
The output of this is: 
is subscriptionOfSubject closed? false
on resolve
is subscriptionOfSubject closed? **false**

**Expected behavior:**
I expect, that the Subscription of the Subject should not close. 

**Actual behavior:**
The Subject closes although it has nothing to do with the Observable Promise (1. code snipped). If I add a callback to the subscribe function of the Promise it works as I expect. (2. code snipped)

**Additional information:**
I am only using the defered promise for this example, in my project I'm using a "normal" promise.

Am I doing something wrong, or could that be a bug?
Thank you in advance.

**Comment by trxcllnt** [comment]
This should be fixed by #2238, but hasn't gone out in a release yet.

**Comment by greenarr0w** [comment]
Thank you for the answer.

**Comment by kwonoj** [comment]
I'm closing this as #2238 included in 5.3.0, expect latest release resolves issue.
